### PR TITLE
Stop loading neurons in NnsProposals

### PR DESCRIPTION
**Pull request by dskloetd**
# Motivation

There used to be a checkbox on the `NnsProposals` page to only show proposals that you can vote on.
This applied a filter on `list_proposals` to only load proposals with ballots matching your neurons.
For this reason we needed to reload the proposals when your neurons changed.
But we no longer have this checkbox (we have the actionable proposals tab instead) so we no longer need:
1. load neurons in `NnsProposals`
2. reload proposals when the neurons change

# Changes

1. Stop loading neurons in `NnsProposals`.
2. Stop listening to `definedNeuronsStore` to reload proposals.

# Tests

1. With the reloading on neurons removed, the tests no longer hit the debounce, so instead we use an unresolved promise to test the skeleton cards and the spinner.
2. Two tests were setting `proposalsFiltersStore` immediate after rendering the component, for no apparently reason. This was causing proposals to be reloaded, making the test more complicated. I've removed the unnecessary setting of `proposalsFiltersStore`.
3. I added a separate test to test that proposals are reloaded when the `proposalsFiltersStore` is set.
4. Remove a work-around which is no longer necessary from `NnsProposalListPo.waitForContentLoaded`.
5. `NnsProposalListPo.waitForContentLoaded` was actually broken because the `.waitForAbsent()` calls weren't awaited. This wasn't noticed because the 1 second delay was enough for the content to load, but fixing the `waitForContentLoaded` method exposed that we were waiting for `NnsProposalListPo` on the actionable proposals page which does not have `NnsProposalListPo`. So I removed this wait from `neurons.spec.ts` and `proposals.spec.ts`.

# Todos

- [ ] Add entry to changelog (if necessary).
not necessary

### CODE REVIEW

File: frontend/src/tests/lib/pages/NnsProposals.spec.ts [hunk 1]
```diff
@@ -111,15 +87,87 @@ describe("NnsProposals", () => {
...
+
+        expect(await po.getSkeletonCardPo().isPresent()).toEqual(true);
+
+        // Let the proposals load.
+        resolveQueryProposals(mockProposals);
+        await runResolvedPromises();
+        expect(await po.getSkeletonCardPo().isPresent()).toEqual(false);
+      });
+
+      it("should reload transactions when proposal filter is set", async () => {
```
Comment (mstrasinskis) [inline frontend/src/tests/lib/pages/NnsProposals.spec.ts:108]: Probably it should reload ~~transactions~~ proposals.

File: frontend/src/tests/lib/pages/NnsProposals.spec.ts [hunk 2]
```diff
@@ -111,15 +87,87 @@ describe("NnsProposals", () => {
...
+        expect(queryProposalsSpy).toBeCalledWith({
+          ...expectedQueryProposalsParams,
+          certified: false,
+        });
+        expect(queryProposalsSpy).toBeCalledWith({
+          ...expectedQueryProposalsParams,
+          certified: true,
+        });
+
+        queryProposalsSpy.mockClear();
```
Comment (mstrasinskis) [inline frontend/src/tests/lib/pages/NnsProposals.spec.ts:145]: Nice!

**Review by mstrasinskis — APPROVED** [review]
Thank you!

File: frontend/src/tests/lib/pages/NnsProposals.spec.ts (hunk 1 above)
Comment (dskloetd) [reply frontend/src/tests/lib/pages/NnsProposals.spec.ts]: Yes, thanks!

**Pull request merged**